agent/eventhandler/task_handler.go

```diff
@@ -256,3 +279,77 @@ func (handler *TaskHandler) SubmitTaskEvents(taskEvents *eventList, client api.E
 		})
 	}
 }
+
+// sendEvent tries to send an event, specified by 'eventToSubmit', of type
+// 'eventType' to ECS
+func sendEvent(sendStatusToECS sendStatusChangeToECS,
+	setChangeSent setStatusSent,
+	eventType string,
+	client api.ECSClient,
+	eventToSubmit *list.Element,
+	stateSaver statemanager.Saver,
+	backoff *utils.SimpleBackoff,
+	taskEvents *eventList) {
+
+	// Extract the wrapped event from the list element
+	event := eventToSubmit.Value.(*sendableEvent)
+	seelog.Infof("TaskHandler, Sending %s change: %s", eventType, event.String())
+	// Try submitting the change to ECS
+	if err := sendStatusToECS(client, event); err != nil {
+		seelog.Errorf("TaskHandler, Unretriable error submitting %s state change [%s]: %v",
+			eventType, event.String(), err)
+		return
+	}
+	// submitted; ensure we don't retry it
+	event.setSent()
+	// Mark event as sent
+	setChangeSent(event)
+	// Update the state file
+	stateSaver.Save()
+	seelog.Debugf("TaskHandler, Submitted container state change: %s", event.String())
+	backoff.Reset()
+	taskEvents.events.Remove(eventToSubmit)
+}
+
+// sendStatusChangeToECS defines a function type for invoking the appropriate ECS state change API
+type sendStatusChangeToECS func(client api.ECSClient, event *sendableEvent) error
+
+// sendContainerStatusToECS invokes the SubmitContainerStateChange API to send a
+// container status change to ECS
+func sendContainerStatusToECS(client api.ECSClient, event *sendableEvent) error {
+	return client.SubmitContainerStateChange(event.containerChange)
+}
+
+// sendTaskStatusToECS invokes the SubmitTaskStateChange API to send a task
+// status change to ECS
+func sendTaskStatusToECS(client api.ECSClient, event *sendableEvent) error {
+	return client.SubmitTaskStateChange(event.taskChange)
+}
+
+// setStatusSent defines a function type to mark the event as sent
+type setStatusSent func(event *sendableEvent)
+
+// setContainerChangeSent sets the event's container change object as sent
+func setContainerChangeSent(event *sendableEvent) {
+	if event.containerChange.Container != nil {
+		event.containerChange.Container.SetSentStatus(event.containerChange.Status)
+	}
+}
+
+// setTaskChangeSent sets the event's task change object as sent
+func setTaskChangeSent(event *sendableEvent) {
+	if event.taskChange.Task != nil {
+		event.taskChange.Task.SetSentStatus(event.taskChange.Status)
+	}
+	for _, container := range event.taskChange.Containers {
+		container.Container.SetSentStatus(event.taskChange.Status.ContainerStatus(container.Container.GetSteadyStateStatus()))
```

Comment:
Why not just `container.Status` instead of `event.taskChange.Status.ContainerStatus(container.Container.GetSteadyStateStatus())`?

Reply:
I missed that we have this in `ContainerStateChange`. I've modified this.

Reply:
Looks like this is still not fixed, did you miss this in the commit?

Reply:
Seems so. Working on it.